include/rocksdb/sst_file_reader.h

```diff
@@ -0,0 +1,91 @@
+//  Copyright (c) 2018-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#ifndef ROCKSDB_LITE
+
+#include <string>
+#include "table/table_reader.h"
+
+namespace rocksdb {
+
+// SstFileReader is used to read sst files.
+class SstFileReader {
+ public:
+  // `file_name` specifies path of the read-only sst file to be accessed.
+  // If `verify_checksum` is true, all data read from underlying storage will be
+  // verified against corresponding checksums.
+  SstFileReader(const std::string& file_name, bool verify_checksum);
+
+  virtual ~SstFileReader() {}
+
+  // Returns a new iterator over this sst file.
+  // The lifetime of return value should be shorter than `SstFileReader`.
+  std::unique_ptr<InternalIterator> NewIterator();
+
+  // Get table properties.
+  Status ReadTableProperties(std::shared_ptr<const TableProperties>* table_properties);
+
+  // Returns init result of `SstFileReader`.
+  // It can be invoked after constructor to tell whether `SstFileReader` is ready to use.
+  Status getStatus() { return init_result_; }
+
+  // Check whether there is corruption in this file.
+  Status VerifyChecksum();
+
+ protected:
+
+  enum SstFileFormat {
+    BlockBased,
+    PlainTable,
+    Unsupported,
+  };
+
+  SstFileReader();
+
+  Status Open(const std::string& file_name, bool verify_checksum);
+
+  // Get the TableReader implementation for the sst file
+  Status GetTableReader(const std::string& file_path);
+
+  // Helper function to call the factory with settings specific to the
+  // factory implementation
+  Status NewTableReader(const ImmutableCFOptions& ioptions,
+                        const EnvOptions& soptions,
+                        const InternalKeyComparator& internal_comparator,
+                        uint64_t file_size,
+                        std::unique_ptr<TableReader>* table_reader);
+
+  virtual Status ReadTableProperties(uint64_t table_magic_number,
+                                     RandomAccessFileReader* file, uint64_t file_size);
+
+  virtual Status SetTableOptionsByMagicNumber(uint64_t table_magic_number);
+  virtual void SetBlockBasedTableOptionsByMagicNumber();
+  virtual void SetPlainTableOptionsByMagicNumber();
+  virtual Status SetOldTableOptions();
+
+  SstFileFormat GetSstFileFormat(uint64_t table_magic_number);
+
+  std::string file_name_;
+  bool verify_checksum_;
+  EnvOptions soptions_;
+
+  // options_ and internal_comparator_ will also be used in NewIterator,
+  // ReadSequential internally (specifically, seek-related operations)
+  Options options_;
+  const ImmutableCFOptions ioptions_;
+  const MutableCFOptions moptions_;
```

Comment:
`MutableCFOptions` doesn't seem to be under `include/rocksdb`, so I'm not sure users can use this class.